Update the default log pattern

tomscut opened the pull request:
Master Issue: [#8298](https://github.com/apache/pulsar/issues/8298)

### Motivation

To make it easy to track problems and resolve them, we should add date for the logs by default.

### Modifications

Updated the default logs pattern, added date.

### Verifying this change

No tests involved.

Before the change:
![before-change](https://user-images.githubusercontent.com/55134131/132304198-6544bcc7-25c1-43eb-adad-518114c1a2b5.jpg)

After the change:
![after-change](https://user-images.githubusercontent.com/55134131/132339368-94b1ace6-7ae2-4275-8536-d6393a325947.jpg)

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (yes / no)
  - The public API: (no)
  - The schema: (no)
  - The default values of configurations: (no)
  - The wire protocol: (no)
  - The rest endpoints: (no)
  - The admin cli options: (no)
  - Anything that affects deployment: (no)

### Documentation

- [ ] doc-required 
- [x] no-need-doc 
- [ ] doc

tomscut commented:
Hi @sijie @hangc0276 @codelipenghui , could you please review the code and approve running workflows? Thanks.

tomscut commented:
Thanks @codelipenghui for your review.

eolivelli reviewed (changes requested):
Overall I agree with this change.
But we should discuss about this on dev@pulsar.apache.org

there are many systems that parse logs and if we change the pattern we are going to break things.
So the community must be informed and agree on this change.

I am setting "Request changes" only because I would like to see that we all agree

tomscut commented:
> Overall I agree with this change.
> But we should discuss about this on [dev@pulsar.apache.org](mailto:dev@pulsar.apache.org)
> 
> there are many systems that parse logs and if we change the pattern we are going to break things.
> So the community must be informed and agree on this change.
> 
> I am setting "Request changes" only because I would like to see that we all agree

Thank you for your suggestion and careful consideration. I will initiate a discussion on dev@pulsar.apache.org.

ivankelly reviewed (changes requested):
Please include the timezone also. 

Rather than a custom format, you can use the log4j2 builtin format.
%d{ISO8601_OFFSET_DATE_TIME_HHMM}

tomscut commented:
> Please include the timezone also.
> 
> Rather than a custom format, you can use the log4j2 builtin format.
> %d{ISO8601_OFFSET_DATE_TIME_HHMM}

Thanks @ivankelly for your review and suggestion. I'm in favor of adding timezone and I will make some change.

tomscut commented:
> Please include the timezone also.
> 
> Rather than a custom format, you can use the log4j2 builtin format.
> %d{ISO8601_OFFSET_DATE_TIME_HHMM}

I added timezone and replaced custom format with ISO8601_OFFSET_DATE_TIME_HHMM. Could please take a look? Thanks a lot.

ivankelly reviewed (approved):
lgtm.

tomscut commented:
> lgtm.

Thanks for you quick reply.

michaeljmarshall reviewed (approved):
LGTM.

I'd like to point out that this change will increase the raw size of the logging output (and given that the logs are verbose already, that will lead to more logging output for large clusters). I don't think that should prevent this change, though.

tomscut commented:
Hi @sijie @merlimat , could you please review this change? Thanks a lot.

eolivelli reviewed (approved):
LGTM

I have dropped the '2.8.2' label because this is an important change and we cannot do it in a point release

tomscut commented:
> LGTM
> 
> I have dropped the '2.8.2' label because this is an important change and we cannot do it in a point release

Thanks @eolivelli for your review. 

I found that ```CPP, Python Tests``` failed, but I did not see the correlation with this change. Could you please help confirm this problem? I'm worried about introducing some problems.

tomscut commented:
Hi @merlimat @Licht-T , could you please look at this problem ```CPP, Python Tests failed```. I'm not sure if the failed test is related to this change and how to fix it. Thanks a lot.

tomscut commented:
> Hi @merlimat @Licht-T , could you please look at this problem `CPP, Python Tests failed`. I'm not sure if the failed test is related to this change and how to fix it. Thanks a lot.

I found the problem caused by this code format and it has been fixed. All checks have passed now.
